com.sap.cloud.lm.sl.cf.process/src/test/java/com/sap/cloud/lm/sl/cf/process/steps/PollServiceInProgessOperationsExecutionTest.java

```diff
@@ -0,0 +1,204 @@
+    @Mock
+    private ServiceOperationGetter serviceOperationGetter;
+    @Mock
+    private ServiceProgressReporter serviceProgressReporter;
+    @Mock
+    private EventsGetter eventsGetter;
+    @Mock
+    private CloudControllerClient client;
+    @Rule
+    public ExpectedException exception = ExpectedException.none();
+    private List<String> serviceNames;
+    private List<ServiceOperationType> servicesOperationTypes;
+    private List<ServiceOperationState> servicesOperationStates;
+    private boolean shouldVerifyStepLogger;
+    private AsyncExecutionState expectedExecutionState;
+    private String expectedExceptionMessage;
+
+    @Before
+    public void setUp() {
+        context.setVariable(Constants.VARIABLE_NAME_SPACE_ID, TEST_SPACE_ID);
+        List<CloudServiceExtended> services = generateCloudServicesExtended();
+        prepareServiceOperationGetter(services);
+        prepareServicesData(services);
+        prepareTriggeredServiceOperations();
+        when(clientProvider.getControllerClient(anyString(), anyString())).thenReturn(client);
+        if (expectedExceptionMessage != null) {
+            exception.expectMessage(expectedExceptionMessage);
+        }
+    }
+
+    private void prepareServiceOperationGetter(List<CloudServiceExtended> services) {
+        for (int i = 0; i < services.size(); i++) {
+            CloudServiceExtended service = services.get(i);
+            ServiceOperationType serviceOperationType = servicesOperationTypes.get(i);
+            ServiceOperationState serviceOperationState = servicesOperationStates.get(i);
+            if (serviceOperationType != null && serviceOperationState != null) {
+                when(serviceOperationGetter.getLastServiceOperation(any(),
+                                                                    eq(service))).thenReturn(new ServiceOperation(serviceOperationType,
+                                                                                                                  "",
+                                                                                                                  serviceOperationState));
+            }
+        }
+    }
+
+    private void prepareTriggeredServiceOperations() {
+        Map<String, ServiceOperationType> trigerredServiceOperations = new HashMap<>();
+        for (int index = 0; index < serviceNames.size(); index++) {
+            String serviceName = serviceNames.get(index);
+            ServiceOperationType serviceOperationType = servicesOperationTypes.get(index);
+            if (serviceOperationType != null) {
+                trigerredServiceOperations.put(serviceName, serviceOperationType);
+            }
+        }
+        StepsUtil.setTriggeredServiceOperations(context, trigerredServiceOperations);
+    }
+
+    private List<CloudServiceExtended> generateCloudServicesExtended() {
+        return serviceNames.stream()
+                           .map(serviceName -> ImmutableCloudServiceExtended.builder()
```

Comment:
Extract this mapping to a method and use it as a method reference

Reply:
Done